--- build/bin/hpm.lua	2016-09-16 16:14:55.610740230 +0700
+++ build/bin/hpm.lua	2016-09-16 16:39:06.868611411 +0700
@@ -1,1 +1,222 @@
+local json = load([===[
+local rA5U=20160728.17
+local Uc06="-[ JSON.lua package by Jeffrey Friedl (http://regex.info/blog/lua/json) version 20160728.17 ]-"local lcBL={VERSION=rA5U,AUTHOR_NOTE=Uc06}local DHPxI="  "
+local dx={pretty=true,align_keys=false,indent=DHPxI}
+local RRuSHnxf={__tostring=function()return"JSON array"end}RRuSHnxf.__index=RRuSHnxf
+local mcYOuT={__tostring=function()return"JSON object"end}mcYOuT.__index=mcYOuT;function lcBL:newArray(iXxD6s)
+return setmetatable(iXxD6s or{},RRuSHnxf)end;function lcBL:newObject(oiY)return
+setmetatable(oiY or{},mcYOuT)end;local function Rr(FsYIVlkf)
+return
+type(FsYIVlkf)=='number'and FsYIVlkf or FsYIVlkf.N end
+local scRP0={__index=isNumber,__tostring=function(HLXS0Q_)return HLXS0Q_.S end,__unm=function(Kw)return
+Rr(Kw)end,__concat=function(nvaIsNv7,vDnoL55)
+return tostring(nvaIsNv7)..tostring(vDnoL55)end,__add=function(xlAK,zr1y)return Rr(xlAK)+Rr(zr1y)end,__sub=function(Hs,jk)return
+Rr(Hs)-Rr(jk)end,__mul=function(qzSFyIO,Z65)
+return Rr(qzSFyIO)*Rr(Z65)end,__div=function(umyCNfj,FT)return Rr(umyCNfj)/Rr(FT)end,__mod=function(YVLXYq,bJfct)return
+Rr(YVLXYq)%Rr(bJfct)end,__pow=function(OhuFpq_N,Dzg)return
+Rr(OhuFpq_N)^Rr(Dzg)end,__lt=function(_4O,C)return Rr(_4O)<Rr(C)end,__eq=function(fLI2zRe,_Fr2YU)return
+Rr(fLI2zRe)==Rr(_Fr2YU)end,__le=function(Xfn,U)return
+Rr(Xfn)<=Rr(U)end}
+function lcBL:asNumber(Ebsw)
+if getmetatable(Ebsw)==scRP0 then return Ebsw elseif
+type(Ebsw)=='table'and
+type(Ebsw.S)=='string'and type(Ebsw.N)=='number'then return setmetatable(Ebsw,scRP0)else
+local UlikV={S=tostring(Ebsw),N=tonumber(Ebsw)}return setmetatable(UlikV,scRP0)end end
+local function AI0R2TQ6(JtAjijkG)
+if JtAjijkG<=127 then return string.char(JtAjijkG)elseif JtAjijkG<=2047 then
+local s=math.floor(JtAjijkG/0x40)local YAtG_LV3=JtAjijkG- (0x40*s)return
+string.char(0xC0+s,0x80+YAtG_LV3)elseif JtAjijkG<=65535 then
+local LfEJbh_=math.floor(JtAjijkG/0x1000)local JD=JtAjijkG-0x1000*LfEJbh_
+local u=math.floor(JD/0x40)local pzDMZwG=JD-0x40*u;LfEJbh_=0xE0+LfEJbh_;u=0x80+u
+pzDMZwG=0x80+pzDMZwG
+if
+
+(LfEJbh_==0xE0 and u<0xA0)or
+(LfEJbh_==0xED and u>0x9F)or(LfEJbh_==0xF0 and u<0x90)or(LfEJbh_==0xF4 and u>0x8F)then return"?"else return string.char(LfEJbh_,u,pzDMZwG)end else local XPoQB=math.floor(JtAjijkG/0x40000)
+local XxJ=JtAjijkG-0x40000*XPoQB;local o5sms=math.floor(XxJ/0x1000)
+XxJ=XxJ-0x1000*o5sms;local JQi1jg=math.floor(XxJ/0x40)local wVzn=XxJ-0x40*JQi1jg
+return string.char(
+0xF0+XPoQB,0x80+o5sms,0x80+JQi1jg,0x80+wVzn)end end
+function lcBL:onDecodeError(pE,RSjapQ,QJf,zC)if RSjapQ then
+if QJf then
+pE=string.format("%s at char %d of: %s",pE,QJf,RSjapQ)else pE=string.format("%s: %s",pE,RSjapQ)end end
+if zC~=nil then pE=pE.." ("..
+lcBL:encode(zC)..")"end
+if self.assert then self.assert(false,pE)else assert(false,pE)end end;lcBL.onDecodeOfNilError=lcBL.onDecodeError
+lcBL.onDecodeOfHTMLError=lcBL.onDecodeError
+function lcBL:onEncodeError(pfZ3SPy_,pDNa2ox6)
+if pDNa2ox6 ~=nil then pfZ3SPy_=pfZ3SPy_..
+" ("..lcBL:encode(pDNa2ox6)..")"end;if self.assert then self.assert(false,pfZ3SPy_)else
+assert(false,pfZ3SPy_)end end
+local function yA(Do6yo7nm,y06X3k,ivnJjrA,d3fMjkg)
+local el=y06X3k:match('^-?[1-9]%d*',ivnJjrA)or y06X3k:match("^-?0",ivnJjrA)if not el then
+Do6yo7nm:onDecodeError("expected number",y06X3k,ivnJjrA,d3fMjkg.etc)end
+local Wu_uIt=ivnJjrA+el:len()local w=y06X3k:match('^%.%d+',Wu_uIt)or""Wu_uIt=Wu_uIt+
+w:len()
+local sgeP=y06X3k:match('^[eE][-+]?%d+',Wu_uIt)or""Wu_uIt=Wu_uIt+sgeP:len()local CM=el..w..sgeP;if
+d3fMjkg.decodeNumbersAsObjects then return lcBL:asNumber(CM),Wu_uIt end
+if
+
+(
+d3fMjkg.decodeIntegerStringificationLength and(el:len()>=d3fMjkg.decodeIntegerStringificationLength or
+sgeP:len()>0))or
+(d3fMjkg.decodeDecimalStringificationLength and
+(
+w:len()>=d3fMjkg.decodeDecimalStringificationLength or sgeP:len()>0))then return CM,Wu_uIt end;local Qlmlet=tonumber(CM)if not Qlmlet then
+Do6yo7nm:onDecodeError("bad number",y06X3k,ivnJjrA,d3fMjkg.etc)end;return Qlmlet,Wu_uIt end
+local function XmVolesU(_,RkGFh6,hw18,nvCiFt7r)if RkGFh6:sub(hw18,hw18)~='"'then
+_:onDecodeError("expected string's opening quote",RkGFh6,hw18,nvCiFt7r.etc)end;local xSebv5Jc=hw18+1
+local mMp=RkGFh6:len()local rDtVf=""
+while xSebv5Jc<=mMp do local vj=RkGFh6:sub(xSebv5Jc,xSebv5Jc)if
+vj=='"'then return rDtVf,xSebv5Jc+1 end
+if vj~='\\'then rDtVf=rDtVf..vj;xSebv5Jc=
+xSebv5Jc+1 elseif RkGFh6:match('^\\b',xSebv5Jc)then rDtVf=rDtVf.."\b"xSebv5Jc=
+xSebv5Jc+2 elseif RkGFh6:match('^\\f',xSebv5Jc)then rDtVf=rDtVf.."\f"xSebv5Jc=
+xSebv5Jc+2 elseif RkGFh6:match('^\\n',xSebv5Jc)then rDtVf=rDtVf.."\n"xSebv5Jc=
+xSebv5Jc+2 elseif RkGFh6:match('^\\r',xSebv5Jc)then rDtVf=rDtVf.."\r"xSebv5Jc=
+xSebv5Jc+2 elseif RkGFh6:match('^\\t',xSebv5Jc)then rDtVf=rDtVf.."\t"xSebv5Jc=
+xSebv5Jc+2 else
+local z=RkGFh6:match('^\\u([0123456789aAbBcCdDeEfF][0123456789aAbBcCdDeEfF][0123456789aAbBcCdDeEfF][0123456789aAbBcCdDeEfF])',xSebv5Jc)
+if z then xSebv5Jc=xSebv5Jc+6;local Zg=tonumber(z,16)
+if
+Zg>=0xD800 and Zg<=0xDBFF then
+local ykRppH=RkGFh6:match('^\\u([dD][cdefCDEF][0123456789aAbBcCdDeEfF][0123456789aAbBcCdDeEfF])',xSebv5Jc)
+if ykRppH then xSebv5Jc=xSebv5Jc+6;Zg=0x2400+ (Zg-0xD800)*0x400+
+tonumber(ykRppH,16)else end end;rDtVf=rDtVf..AI0R2TQ6(Zg)else rDtVf=rDtVf..
+RkGFh6:match('^\\(.)',xSebv5Jc)xSebv5Jc=xSebv5Jc+2 end end end
+_:onDecodeError("unclosed string",RkGFh6,hw18,nvCiFt7r.etc)end
+local function eZ0l3ch(WQ6,y36Aetn)local iPL3B4cr,GI2hz6SK=WQ6:find("^[ \n\r\t]+",y36Aetn)if GI2hz6SK then
+return GI2hz6SK+1 else return y36Aetn end end;local W_63_9
+local function h9dyA_4T(Oh,PG,n,O)if PG:sub(n,n)~='{'then
+Oh:onDecodeError("expected '{'",PG,n,O.etc)end;local N5UjTN=eZ0l3ch(PG,n+1)local qLH5=Oh.strictTypes and
+Oh:newObject{}or{}if
+PG:sub(N5UjTN,N5UjTN)=='}'then return qLH5,N5UjTN+1 end
+local tE=PG:len()
+while N5UjTN<=tE do local VcV0EuD,pX4gCR=XmVolesU(Oh,PG,N5UjTN,O)
+N5UjTN=eZ0l3ch(PG,pX4gCR)if PG:sub(N5UjTN,N5UjTN)~=':'then
+Oh:onDecodeError("expected colon",PG,N5UjTN,O.etc)end
+N5UjTN=eZ0l3ch(PG,N5UjTN+1)local gad4ZcL,pX4gCR=W_63_9(Oh,PG,N5UjTN,O)qLH5[VcV0EuD]=gad4ZcL
+N5UjTN=eZ0l3ch(PG,pX4gCR)local dk=PG:sub(N5UjTN,N5UjTN)
+if dk=='}'then return qLH5,N5UjTN+1 end;if PG:sub(N5UjTN,N5UjTN)~=','then
+Oh:onDecodeError("expected comma or '}'",PG,N5UjTN,O.etc)end
+N5UjTN=eZ0l3ch(PG,N5UjTN+1)end;Oh:onDecodeError("unclosed '{'",PG,n,O.etc)end
+local function oh(E,OO,y,cR6rJlAl)if OO:sub(y,y)~='['then
+E:onDecodeError("expected '['",OO,y,cR6rJlAl.etc)end;local M6ilzGJ=eZ0l3ch(OO,y+1)local iW6CD=E.strictTypes and
+E:newArray{}or{}if
+OO:sub(M6ilzGJ,M6ilzGJ)==']'then return iW6CD,M6ilzGJ+1 end
+local wZdg=1;local BaX=OO:len()
+while M6ilzGJ<=BaX do
+local SJsW11k,Ki1HJT=W_63_9(E,OO,M6ilzGJ,cR6rJlAl)iW6CD[wZdg]=SJsW11k;wZdg=wZdg+1;M6ilzGJ=eZ0l3ch(OO,Ki1HJT)
+local wjim8xCV=OO:sub(M6ilzGJ,M6ilzGJ)if wjim8xCV==']'then return iW6CD,M6ilzGJ+1 end;if
+OO:sub(M6ilzGJ,M6ilzGJ)~=','then
+E:onDecodeError("expected comma or '['",OO,M6ilzGJ,cR6rJlAl.etc)end
+M6ilzGJ=eZ0l3ch(OO,M6ilzGJ+1)end
+E:onDecodeError("unclosed '['",OO,y,cR6rJlAl.etc)end
+W_63_9=function(E,QLam,qTDt,v)qTDt=eZ0l3ch(QLam,qTDt)
+if qTDt>QLam:len()then E:onDecodeError("unexpected end of string",QLam,
+nil,v.etc)end
+if QLam:find('^"',qTDt)then return XmVolesU(E,QLam,qTDt,v.etc)elseif
+QLam:find('^[-0123456789 ]',qTDt)then return yA(E,QLam,qTDt,v)elseif QLam:find('^%{',qTDt)then
+return h9dyA_4T(E,QLam,qTDt,v)elseif QLam:find('^%[',qTDt)then return oh(E,QLam,qTDt,v)elseif QLam:find('^true',qTDt)then return true,
+qTDt+4 elseif QLam:find('^false',qTDt)then return false,qTDt+5 elseif
+QLam:find('^null',qTDt)then return nil,qTDt+4 else
+E:onDecodeError("can't parse JSON",QLam,qTDt,v.etc)end end
+function lcBL:decode(Ta,u,nArcvQl)if type(nArcvQl)~='table'then nArcvQl={}end;if u~=nil then
+nArcvQl.etc=u end;if
+type(self)~='table'or self.__index~=lcBL then
+lcBL:onDecodeError("JSON:decode must be called in method format",nil,nil,nArcvQl.etc)end
+if Ta==nil then
+self:onDecodeOfNilError(string.format("nil passed to JSON:decode()"),
+nil,nil,nArcvQl.etc)elseif type(Ta)~='string'then
+self:onDecodeError(string.format("expected string argument to JSON:decode(), got %s",type(Ta)),
+nil,nil,nArcvQl.etc)end;if Ta:match('^%s*$')then return nil end;if Ta:match('^%s*<')then
+self:onDecodeOfHTMLError(string.format("html passed to JSON:decode()"),Ta,
+nil,nArcvQl.etc)end
+if
+Ta:sub(1,1):byte()==0 or
+(Ta:len()>=2 and Ta:sub(2,2):byte()==0)then
+self:onDecodeError("JSON package groks only UTF-8, sorry",Ta,nil,nArcvQl.etc)end;if nArcvQl.decodeNumbersAsObjects==nil then
+nArcvQl.decodeNumbersAsObjects=self.decodeNumbersAsObjects end;if
+nArcvQl.decodeIntegerStringificationLength==nil then
+nArcvQl.decodeIntegerStringificationLength=self.decodeIntegerStringificationLength end;if
+nArcvQl.decodeDecimalStringificationLength==nil then
+nArcvQl.decodeDecimalStringificationLength=self.decodeDecimalStringificationLength end
+local h6Ub7U,Gm=pcall(W_63_9,self,Ta,1,nArcvQl)if h6Ub7U then return Gm else
+if self.assert then self.assert(false,Gm)else assert(false,Gm)end;return nil,Gm end end
+local function DZXGTh(YKA7cU)
+if YKA7cU=="\n"then return"\\n"elseif YKA7cU=="\r"then return"\\r"elseif YKA7cU=="\t"then return"\\t"elseif YKA7cU=="\b"then
+return"\\b"elseif YKA7cU=="\f"then return"\\f"elseif YKA7cU=='"'then return'\\"'elseif YKA7cU=='\\'then return'\\\\'else return
+string.format("\\u%04x",YKA7cU:byte())end end
+local Su9Koz='['..'"'..
+'%\\'..'%z'..'\001'..'-'..'\031'..']'local Uk7e=AI0R2TQ6(0x2028)local KwQCk_G=AI0R2TQ6(0x2029)
+local function ptZa(mCsewfX,yY)
+local Xf=mCsewfX:gsub(Su9Koz,DZXGTh)if yY.stringsAreUtf8 then
+Xf=Xf:gsub(Uk7e,'\\u2028'):gsub(KwQCk_G,'\\u2029')end;return'"'..Xf..'"'end
+local function PEqsd(UlFdiZ7v,U,wFeA)local JQgI={}local N={}local fs52REi=false;local PUNkgaiM
+for X in pairs(U)do
+if type(X)=='string'then
+table.insert(JQgI,X)elseif type(X)=='number'then table.insert(N,X)
+if X<=0 or X>=math.huge then
+fs52REi=true elseif not PUNkgaiM or X>PUNkgaiM then PUNkgaiM=X end else
+UlFdiZ7v:onEncodeError("can't encode table with a key of type "..type(X),wFeA)end end
+if#JQgI==0 and not fs52REi then
+if#N>0 then return nil,PUNkgaiM elseif
+tostring(U)=="JSON array"then return nil elseif tostring(U)=="JSON object"then return{}else return nil end end;table.sort(JQgI)local s6FbB
+if#N>0 then if UlFdiZ7v.noKeyConversion then
+UlFdiZ7v:onEncodeError("a table with both numeric and string keys could be an object or array; aborting",wFeA)end
+s6FbB={}for dc61,aguhyl in pairs(U)do s6FbB[dc61]=aguhyl end
+table.sort(N)
+for p,gOPDv in ipairs(N)do local aSdZU3=tostring(gOPDv)
+if s6FbB[aSdZU3]==nil then
+table.insert(JQgI,aSdZU3)s6FbB[aSdZU3]=U[gOPDv]else
+UlFdiZ7v:onEncodeError(
+"conflict converting table with mixed-type keys into a JSON object: key "..gOPDv.." exists both as a string and a number.",wFeA)end end end;return JQgI,nil,s6FbB end;local iSj
+function iSj(YKDL,oFyb6OLp,oGdh_mv,WjvvK,TASVwBgU,KjUncMB,XkT)
+if oFyb6OLp==nil or
+(not XkT and TASVwBgU and TASVwBgU.null and oFyb6OLp==
+TASVwBgU.null)then return'null'elseif
+type(oFyb6OLp)=='string'then return ptZa(oFyb6OLp,TASVwBgU)elseif type(oFyb6OLp)=='number'then
+if
+oFyb6OLp~=oFyb6OLp then return"null"elseif oFyb6OLp>=math.huge then return"1e+9999"elseif oFyb6OLp<=-math.huge then
+return"-1e+9999"else return tostring(oFyb6OLp)end elseif type(oFyb6OLp)=='boolean'then return tostring(oFyb6OLp)elseif type(oFyb6OLp)~=
+'table'then
+YKDL:onEncodeError("can't convert "..type(oFyb6OLp).." to JSON",WjvvK)elseif getmetatable(oFyb6OLp)==scRP0 then return tostring(oFyb6OLp)else
+local c3dr=oFyb6OLp;if type(TASVwBgU)~='table'then TASVwBgU={}end;if type(KjUncMB)~=
+'string'then KjUncMB=""end
+if oGdh_mv[c3dr]then
+YKDL:onEncodeError("table "..
+tostring(c3dr).." is a child of itself",WjvvK)else oGdh_mv[c3dr]=true end;local NGH;local tIc,MD2O,HQ=PEqsd(YKDL,c3dr,WjvvK)
+if MD2O then local cng={}for lE=1,MD2O do
+table.insert(cng,iSj(YKDL,c3dr[lE],oGdh_mv,WjvvK,TASVwBgU,KjUncMB))end
+if TASVwBgU.pretty then NGH="[ "..
+table.concat(cng,", ").." ]"else NGH="["..
+table.concat(cng,",").."]"end elseif tIc then local nI2F0id=HQ or c3dr
+if TASVwBgU.pretty then local N4aMD_P={}local pCi=0
+for lNOqUk8,YAnZNei in ipairs(tIc)do
+local h8YWR44E=iSj(YKDL,tostring(YAnZNei),oGdh_mv,WjvvK,TASVwBgU,KjUncMB,true)
+if TASVwBgU.align_keys then pCi=math.max(pCi,#h8YWR44E)end;table.insert(N4aMD_P,h8YWR44E)end
+local NzeoQJ=KjUncMB..tostring(TASVwBgU.indent or"")
+local AwGfFV=NzeoQJ..string.rep(" ",pCi)..
+(TASVwBgU.align_keys and"  "or"")
+local wCRY="%s%"..string.format("%d",pCi).."s: %s"local d0uKSVw1={}
+for VF,fTrMe in ipairs(tIc)do
+local ypDndT8=iSj(YKDL,nI2F0id[fTrMe],oGdh_mv,WjvvK,TASVwBgU,AwGfFV)
+table.insert(d0uKSVw1,string.format(wCRY,NzeoQJ,N4aMD_P[VF],ypDndT8))end;NGH="{\n"..
+table.concat(d0uKSVw1,",\n").."\n"..KjUncMB.."}"else local MV65={}
+for Y3D66Ym9,q in
+ipairs(tIc)do
+local PhJ=iSj(YKDL,nI2F0id[q],oGdh_mv,WjvvK,TASVwBgU,KjUncMB)
+local h=iSj(YKDL,tostring(q),oGdh_mv,WjvvK,TASVwBgU,KjUncMB,true)
+table.insert(MV65,string.format("%s:%s",h,PhJ))end;NGH="{"..table.concat(MV65,",").."}"end else NGH="[]"end;oGdh_mv[c3dr]=false;return NGH end end
+function lcBL:encode(j2K,r8hgwQ,_6U)if type(self)~='table'or self.__index~=lcBL then
+lcBL:onEncodeError("JSON:encode must be called in method format",r8hgwQ)end;if
+type(_6U)~='table'then _6U={}end;return iSj(self,j2K,{},r8hgwQ,_6U)end
+function lcBL:encode_pretty(GLSzBQs,c,xg)if type(self)~='table'or self.__index~=lcBL then
+lcBL:onEncodeError("JSON:encode_pretty must be called in method format",c)end;if
+type(xg)~='table'then xg=dx end;return iSj(self,GLSzBQs,{},c,xg)end;function lcBL.__tostring()return"JSON encode/decode package"end
+lcBL.__index=lcBL
+function lcBL:new(Id2KoP_G)local Y2or={}if Id2KoP_G then
+for zN8ASHV5,iju in pairs(Id2KoP_G)do Y2or[zN8ASHV5]=iju end end
+return setmetatable(Y2or,lcBL)end;return lcBL:new()
+]===])()
 semver = (() ->
